tests/acceptance/job/basic-layout-test.js

```diff
@@ -63,6 +63,211 @@ test('visiting job-view', function (assert) {
   });
 });
 
+
+test('visiting a job in created(received) state', function (assert) {
+  let repo = server.create('repository', { slug: 'travis-ci/travis-web' }),
+    branch = server.create('branch', { name: 'acceptance-tests' });
+
+  let  gitUser = server.create('git-user', { name: 'Mr T' });
+  let commit = server.create('commit', { author: gitUser, committer: gitUser, branch: 'acceptance-tests', message: 'This is a message', branch_is_default: true });
+
+  let request = server.create('request');
+  server.create('message', {
+    request,
+    level: 'info',
+    key: 'group',
+    code: 'flagged',
+    args: {
+      given: 'group'
+    }
+  });
+
+  let user = server.create('user', {
+    name: 'Mr T',
+    avatar_url: '/images/favicon-gray.png'
+  });
+
+  let build = server.create('build', { repository: repo, state: 'created', createdBy: user, commit, branch, request });
+  let job = server.create('job', { number: '1234.1', repository: repo, state: 'created', build, commit });
+  commit.job = job;
+
+  job.save();
+  commit.save();
+
+  server.create('log', { id: job.id });
+
+  visit('/travis-ci/travis-web/jobs/' + job.id);
+  waitForElement('#log > .log-line');
+  andThen(() => {
+    assert.equal(document.title, 'Job #1234.1 - travis-ci/travis-web - Travis CI');
+
+    assert.equal($('head link[rel=icon]').attr('href'), getFaviconUri('yellow'), 'expected the favicon data URI to match the one for created');
+
+    assert.equal(jobPage.branch, 'acceptance-tests', 'displays the branch');
+    assert.equal(jobPage.message, 'Running acceptance-tests This is a message', 'displays message');
+    assert.equal(jobPage.state, '#1234.1 received', 'displays build number');
+
+    assert.equal(jobPage.createdBy.href, '/testuser');
+    assert.equal(jobPage.createdBy.text, 'Mr T');
+    assert.ok(jobPage.createdBy.avatarSrc.startsWith('/images/favicon-gray.png'));
+
+    assert.notOk(jobPage.hasTruncatedLog);
+    assert.ok(jobPage.hasWaitingStates);
```

Comment:
I think you can ignore `hasTruncatedLog`, since the log isn’t even showing.

I noticed that there’s no `assert.notOk(jobPage.hasWaitingStates)` anywhere. When adding a boolean check like this, there should always be at least one assertion for both `true` and `false` values. So you could add a check when the job is `started` or the like that `jobPage.hasWaitingStates` is false.

Reply:
(I did this lower down where I left a comment on the structure of the page object)